### PR TITLE
Use transform to optimize MLP

### PR DESCRIPTION
**Pull request by chelini**
Using transform we can have a better control over fusion and we can generate only 5 loops. 1 loop for each layer (4 layers) + an outermost
one. Bufferization still remains a problem (see issue: #95).

### CODE REVIEW

File: lib/TPP/TransformDialectInterpreter.cpp [hunk 1]
```diff
@@ -38,9 +38,27 @@ struct TransformDialectInterpreter
   }
 };
 
+struct TransformDropSchedulePass
```
Comment (rengolin) [inline lib/TPP/TransformDialectInterpreter.cpp:41]: I'm surprised this isn't done automatically by the pass, or that there isn't this functionality in MLIR proper...

File: include/TPP/Dialect/LinalgX/TransformOps/LinalgXTransformOps.td [hunk 2]
```diff
@@ -207,4 +207,39 @@ def MapLinalgToTppOp : Op<Transform_Dialect, "structured.map_linalg_to_tpp", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// FoldUnitExtentDimsOp
```
Comment (rengolin) [inline include/TPP/Dialect/LinalgX/TransformOps/LinalgXTransformOps.td:211]: I don't understand why this is in `LinalgX` and not on its own dialect. While this operates on `linalg` ops, it's not an extension to the `linalg` dialect.

**Review by rengolin — APPROVED** [review]
Looks good to me. My comments are for future work only.

**Review by adam-smnk — APPROVED** [review]

File: lib/TPP/TransformDialectInterpreter.cpp (hunk 1 above)
Comment (chelini) [reply lib/TPP/TransformDialectInterpreter.cpp:41]: It is not exposed, as Transform is still in test-mode. Something similar is done in IREE too.

**Pull request merged**